test/index.jl

```diff
@@ -360,6 +360,17 @@ end
     @test DataFrames.parentcols(SubIndex(i3, Not(Not(r"")))) == 1:2
 end
 
+@testset "Not indexing with columns that don't exist" begin
```

Comment:
The tests should cover every possible combination types of inputs for `Not`. So these should be:
`Bool`, `Integer`, `AbstractVector{Int}`, `AbstractRange{Int}`, `AbstractVector{<:Integer}`, `AbstractRange{<:Integer}`, `AbstractRange{Bool}`, `AbstractVector{Bool}`, `Colon`, `Not` (nested), `Between`, `All`, `Regex`, `Symbol`, `AbstractString`, `AbstractVector{Symbol}`, `AbstractVector{<:AbstractString}`, `AbstractVector{Any}` (this one testing different combinations allowed in the implementation)

Reply:
Okay I've implemented the functionality that throws errors for integer values. 

`Not()` is way under-tested in general. So I will add tests normal `Not` behavior (columns in the data frame) soon.

Reply:
> Not() is way under-tested in general.

It was not tested before, because we had a very simple logic of processing it before. Since now we deviate from the "default" we should test all cases.

Reply:
let me know if there is anything you would like to change about the current block of tests because they are a bit obtuse and there is going to be a big test diff anyways.